tedana/selection/component_selector.py

```diff
@@ -217,7 +217,14 @@ def validate_tree(tree):
 class ComponentSelector:
     """Load and classify components based on a specified ``tree``."""
 
-    def __init__(self, tree, component_table, cross_component_metrics={}, status_table=None):
+    def __init__(
+        self,
+        tree,
+        component_table,
+        cross_component_metrics={},
+        status_table=None,
+        external_regressor_dict=None,
```

Comment:
I don't think the external regressor config needs to be passed in here.

Reply:
The reason it's passed here is so that the `LGR.report` line for the regressor model appears along with the tree description. This wouldn't be an issue if the regressor model was in the decision tree json, but I think that would cause other issues... still thinking about this.

Reply:
To move everything into the decision tree json file, I think I'd need move a light version of https://github.com/ME-ICA/tedana/pull/969 into this PR. This isn't a bad thing, but the only way to get this to work would be to initialize the component selector and read in the external regressor fields much earlier in the code. There are benefits to this, but it would expand this PR beyond just adding correlations. I might still give it a try, but wanted to bring this up.

Reply:
Done